src/Microsoft.Extensions.SecretManager.Tools/Program.cs

```diff
@@ -78,36 +69,62 @@ internal int RunInternal(params string[] args)
                 return 2;
             }
 
-            if (options.IsVerbose)
-            {
-                CommandOutputProvider.LogLevel = LogLevel.Debug;
-            }
+            var reporter = CreateReporter(options.IsVerbose);
 
             string userSecretsId;
             try
             {
-                userSecretsId = ResolveId(options);
+                userSecretsId = ResolveId(options, reporter);
             }
             catch (Exception ex) when (ex is InvalidOperationException || ex is FileNotFoundException)
             {
-                _logger.LogError(ex.Message);
+                reporter.Error(ex.Message);
                 return 1;
             }
 
-            var store = new SecretsStore(userSecretsId, Logger);
-            var context = new Internal.CommandContext(store, Logger, _console);
+            var store = new SecretsStore(userSecretsId, reporter);
+            var context = new Internal.CommandContext(store, reporter, _console);
             options.Command.Execute(context);
             return 0;
         }
 
-        internal string ResolveId(CommandLineOptions options)
+        private IReporter CreateReporter(bool verbose)
+        {
+            return new ReporterBuilder()
+                .WithConsole(_console)
+                .Verbose(f =>
+                {
+                    f.When(() => verbose);
+                    if (!_console.IsOutputRedirected)
+                    {
+                        f.WithColor(ConsoleColor.DarkGray);
+                    }
```

Comment:
It's output is colorless. Users might pipe commands to a file. We want to avoid spitting out the ANSI color codes.